From 7b27d13a395c4acb460c7e81fc20ae01a8c763cf Mon Sep 17 00:00:00 2001
From: Matthias Kretz <kretz@kde.org>
Date: Tue, 14 Jun 2016 09:09:35 +0200
Subject: [PATCH] Drop AliasStrategy::Union specialization for GCC6

GCC 6 refuses to compile this implementation strategy since it requires
non-standard `EntryType m[]` to keep the ABI.

Refs: gh-125
Signed-off-by: Matthias Kretz <kretz@kde.org>
---
 common/storage.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/storage.h b/common/storage.h
index 258fde9..b995161 100644
--- a/common/storage.h
+++ b/common/storage.h
@@ -145,6 +145,8 @@ using DefaultStrategy =
 template <typename ValueType, size_t Size, typename Strategy = DefaultStrategy>
 class Storage;
 
+#if !defined Vc_GCC || Vc_GCC < 0x60000
+// GCC 6 forbids `EntryType m[]` altogether
 template <typename ValueType, size_t Size>
 class Storage<ValueType, Size, AliasStrategy::Union>
 {
@@ -192,6 +194,7 @@ class Storage<ValueType, Size, AliasStrategy::Union>
 #endif
     } data;
 };
+#endif
 
 template <typename ValueType, size_t Size>
 class Storage<ValueType, Size, AliasStrategy::MayAlias>
